pygls/protocol.py

```diff
@@ -509,7 +509,16 @@ def bf_initialize(self, params: InitializeParams):
 
         client_capabilities = params.capabilities
         root_uri = params.rootUri
-        root_path = params.rootPath
+
```

Comment:
`rootUri` is mandatory so I believe server should fail if it's not provided (it should at least be null)

Reply:
That is a good point, though, since the suggested test would have neither `rootUri` or `rootPath`. 

`rootUri` is derived from `rootPath` a little further below in the code:
```
if root_uri is None:
    root_uri = from_fs_path(root_path) if root_path is not None else ''
```
It may be that instead of setting `rootUri` to empty string, we should keep it as None, in this case. @danixeee Any thoughts?

Reply:
I would go with `getattr(params, 'workspaceFolders', [])`. For me it's more readable, especially if we will have few `try-except` blocks in a row.

PS: @yorodm thank you for your contribution.

Reply:
@danixee @augb Thanks to you guys for the framework. I'm building a Python LSP server with it and it's been wonderful to work with

Reply:
@yorodm Glad to hear it. When you are ready, please submit a PR to update [Implementations.md](https://github.com/openlawlibrary/pygls/blob/master/Implementations.md).